src/components/Admin/Admin.js

```diff
@@ -34,109 +35,145 @@ class Admin extends React.Component {
     super(props);
 
     this.state = {
-      index: 0
+      index: 0,
+      showModal: false
     };
 
+    this.closeModal = this.closeModal.bind(this);
+    this.openModal = this.openModal.bind(this);
     this.handleTabChanged = this.handleTabChanged.bind(this);
+    this.restartPipelineAndCloseModal = this.restartPipelineAndCloseModal.bind(this);
+  }
+
+  closeModal() {
+    this.setState({ showModal: false });
+  }
+
+  openModal() {
+    this.setState({ showModal: true });
   }
 
   handleTabChanged(index, last) {
-    this.setState({ index: index });
+    this.setState({ index });
+  }
+
+  restartPipelineAndCloseModal() {
+    this.props.flux.actions.ADMIN.restart_pipeline();
+    this.closeModal();
   }
 
   render() {
     return (
-      <div className="container-fluid">
-        <div className="col-lg-12">
-          <div className="panel panel-primary" style={styles.container.panel}>
-            <div className="panel-body">
-                <div className="row adminContainer">
-                <Tabs
-                  onSelect={this.handleTabChanged}
-                  selectedIndex={this.state.index}>
-                  <TabList>
-                    <Tab>Site Settings</Tab>
-                    <Tab>Watchlist</Tab>
-                    <Tab>Geofence / Monitored places</Tab>
-                    <Tab>Event Import</Tab>
-                    <Tab>Trusted Sources</Tab>
-                    <Tab>Blacklisted Terms</Tab>
-                    <Tab>Streams</Tab>
-                  </TabList>
-                  <TabPanel>
-                    <h2>Settings</h2>
-                    {
-                      this.props.settings && this.props.settings.properties && this.state.index === SETTINGS_TAB ?
-                      <AdminSettings {...this.props}
-                        index={this.state.index}
-                        siteSettings={this.props.settings}
-                      />
-                      : undefined
-                    }
-                  </TabPanel>
-                  <TabPanel>
-                    <h2>Watchlist</h2>
+      <div>
+        <div className="container-fluid">
+          <div className="col-lg-12">
+            <div className="panel panel-primary" style={styles.container.panel}>
```

Comment:
Would it be possible to replace this explicit markup with react-bootstrap's [Panel](https://react-bootstrap.github.io/components.html#panels) component?

Reply:
It would, but I'd rather do this in a different PR, since I think there's a lot of other bootstrap components in there that we can replace with react-bootstrap's.